src/Data/List/Base.agda

```diff
@@ -292,6 +294,20 @@ span P? (x ∷ xs) with does (P? x)
 break : ∀ {P : Pred A p} → Decidable P → List A → (List A × List A)
 break P? = span (∁? P?)
 
+nub-filter : ∀ {R : Rel A p} → B.Decidable R → A → List A → List A
+nub-filter R? y [] = []
+nub-filter R? y (x ∷ xs) with does (R? y x)
+... | true  = nub-filter R? y xs
+... | false = x ∷ nub-filter R? x xs
+
+nub : ∀ {R : Rel A p} → B.Decidable R → List A → List A
+nub R? []       = []
+nub R? (x ∷ xs) = x ∷ nub-filter R? x xs
```

Comment:
Could we implement `nub` (or whatever we end up calling it!) as follows:
```agda
nub : ∀ {R : Rel A p} → B.Decidable R → List A → List A
nub R? []           = []
nub R? (x ∷ [])     = x ∷ []
nub R? (x ∷ y ∷ xs) with does (R? x y) | nub R? (y ∷ xs)
... | true  | nxs = nxs
... | false | nxs = x ∷ nxs
```
which would remove the need to have `nub-filter` and all it's own associated lemmas? (The nastiness of with abstracting the recursive call is needed due to termination checking getting less powerful in the upcoming release of Agda 2.6.1, see agda/agda#3604)

Reply:
My reading of the agda/agda#3604 was that this used to be allowed but now it will be rejected by termination checker?

At least for recursive calls the agda though that y ∷ xs ≮ x ∷ y ∷ xs and failed the termination checker. My reading of agda/agda#3604 is that this will be true for with abstraction as well?

Reply:
So it's the following that used to be allowed but isn't.
```agda
nub : ∀ {R : Rel A p} → B.Decidable R → List A → List A
nub R? []           = []
nub R? (x ∷ [])     = x ∷ []
nub R? (x ∷ y ∷ xs) with does (R? x y)
... | true  | nxs = nub R? (y ∷ xs)
... | false | nxs = x ∷ nub R? (y ∷ xs)
```
My variant that is listed above is allowed, because the recursive call comes before the with abstraction. Indeed I've tested it out on the release candidate for `2.6.1` and it's passes termination checking.